github/issues_test.go

```diff
@@ -281,7 +286,24 @@ func TestIssuesService_Lock(t *testing.T) {
 		w.WriteHeader(http.StatusNoContent)
 	})
 
-	if _, err := client.Issues.Lock(context.Background(), "o", "r", 1); err != nil {
+	if _, err := client.Issues.Lock(context.Background(), "o", "r", 1, nil); err != nil {
+		t.Errorf("Issues.Lock returned error: %v", err)
+	}
+}
+
+func TestIssuesService_LockWithReason(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/issues/1/lock", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "PUT")
+
```

Comment:
Can you please add a `testHeader` here?
That makes it easier to maintain so that when the header is later removed, it is clear that this needs cleaning up here as well.